samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -345,86 +347,34 @@ private     RenderedTemplate    getRenderedTemplate             (String template
 
         return headers;
     }
-
-    // loadTemplate from ValidationLogicImpl.java
-    /**
-     * Load and register one or more email templates (contained in the given
-     * .xml file) with the email template service
-     *
-     * @param templateFileName - the name of the .xml file to load
-     * @param templateRegistrationString - the key (name) of the template to be saved to the service
-     */
-    @SuppressWarnings("unchecked")
-    private     void                loadTemplate                    (String templateFileName, String templateRegistrationString) {
-        LOG.info(this + " loading template " + templateFileName);
-
-        SecurityAdvisor yesMan = new SecurityAdvisor() {
-            public SecurityAdvice isAllowed( String userId, String function, String reference ) {
-                return SecurityAdvice.ALLOWED;
-            }
-        };
-
-        try {
-            // Push the yesMan SA on the stack and perform the necessary actions
-            securityService.pushAdvisor(yesMan);
-
-            // Load up the resource as an input stream
-            InputStream input = SamigoETSProviderImpl.class.getClassLoader().getResourceAsStream("" + templateFileName);
-            if ( input == null ) {
-                LOG.error( "Could not load resource from '" + templateFileName + "'. Skipping ..." );
-            } else {
-                // Parse the XML, get all the child templates
-                Document document = new SAXBuilder().build( input );
-                List<?> childTemplates = document.getRootElement().getChildren( "emailTemplate" );
-                Iterator<?> iter = childTemplates.iterator();
-
-                // Create and register a template with the service for each one found in the XML file
-                while ( iter.hasNext() ) {
-                    xmlToTemplate( (Element) iter.next(), templateRegistrationString );
-                }
-            }
-        } catch (JDOMException | IOException e) {
-            LOG.error("loadTemplate error: ", e);
-        } finally { // Pop the yesMan SA off the stack (remove elevated permissions)
-            securityService.popAdvisor(yesMan);
+	
+    @Override
+    public void notifyAutoSubmitFailures(int count)
+    {
+        boolean notifyOn = serverConfigurationService.getBoolean(SamigoConstants.SAK_PROP_AUTO_SUBMIT_ERROR_NOTIFICATION_ENABLED, false);
+        if (!notifyOn)
+        {
+            return;
         }
-    }
+        String supportAddress = StringUtils.defaultIfBlank(serverConfigurationService.getString(SamigoConstants.SAK_PROP_SUPPORT_EMAIL_ADDRESS), fromAddress);
+        String toAddress = StringUtils.defaultIfBlank(serverConfigurationService.getString(SamigoConstants.SAK_PROP_AUTO_SUBMIT_ERROR_NOTIFICATION_TO_ADDRESS), supportAddress);
```

Comment:
I think @plukasew has a point here. The JavaDoc clearly states that it *will* return null in the second situation (where you have a property defined without a value). In that case, the code will not return the default value you've given it. It will call StringUtils.trimToNull() on an empty string, and then returns it.

I'm not sure why we would ever want to return `null` in this case. Perhaps the correct path forwards is to fix this in SCS.getString(String name, String dflt). However, doing so may break any existing code which is (for whatever reason) expecting this to return null.